Change home page UI

Manas-Kenge opened the pull request:
## Issue number

- Fixes #1787

## Description

This is the first draft of the UI. Any suggestions are welcome.

## Add screenshots or links to a preview of the changes

![image](https://github.com/user-attachments/assets/8e097166-5feb-463e-9ad3-f317282db227)


<!-- readthedocs-preview plone6 start -->
----
📚 Documentation preview 📚: https://plone6--1875.org.readthedocs.build/

<!-- readthedocs-preview plone6 end -->

Manas-Kenge committed:
change UI

Manas-Kenge committed:
add icons

Manas-Kenge commented:
@stevepiercy could you review this?

Manas-Kenge committed:
add reference guide

Manas-Kenge commented:
As per https://github.com/plone/documentation/issues/1787#issue-2684784725 added a reference guide page, not much in it but it's a start

stevepiercy commented:
@Manas-Kenge I'm super pleased with the result of your work. Thank you!

Would you please delete the text "This documentation is for:" and its bullet points? I think it's just visual noise and not particularly helpful.

Since this is a Big Freaking Deal™, I've requested feedback from the broader community. There may be a lot of discussion and opinions. I'll respond to feedback from other people, and provide further direction, if needed.

I'll do a more thorough review when I have more time.

sneridagh reviewed (dismissed):
LGTM!

Manas-Kenge committed:
Remove bullet points

Manas-Kenge commented:
Thanks for the feedback, @stevepiercy ! I've removed that section. Let me know if there's anything else.

stevepiercy commented on docs/reference-guide/index.md:
```diff
@@ -0,0 +1,27 @@
+    "description": "Plone Reference guide"
+    "property=og:description": "Plone Reference guide"
+    "property=og:title": "Reference guide"
+    "keywords": "Plone 6, reference guide"
```
```suggestion
    "keywords": "Plone, reference, guide"
```

stevepiercy commented on docs/reference-guide/index.md:
```diff
@@ -0,0 +1,27 @@
+  html_meta:
+    "description": "Plone Reference guide"
+    "property=og:description": "Plone Reference guide"
+    "property=og:title": "Reference guide"
```
```suggestion
    "property=og:title": "Plone reference guide"
```

stevepiercy commented on docs/reference-guide/index.md:
```diff
@@ -0,0 +1,27 @@
+---
+myst:
+  html_meta:
+    "description": "Plone Reference guide"
```
```suggestion
    "description": "Plone reference guide"
```

stevepiercy commented on docs/reference-guide/index.md:
```diff
@@ -0,0 +1,27 @@
+myst:
+  html_meta:
+    "description": "Plone Reference guide"
+    "property=og:description": "Plone Reference guide"
```
```suggestion
    "property=og:description": "Plone reference guide"
```

stevepiercy commented on docs/reference-guide/index.md:
```diff
@@ -0,0 +1,27 @@
+
+# Reference guide
+
+This reference guide details functions, modules, and objects included in Plone, describing what they are and what they do. For learning how to use Plone, see the [complete documentation](https://6.docs.plone.org/index.html).
```
```suggestion
This reference guide details application programming interfaces (APIs), functions, modules, and objects included in Plone, describing what they are and what they do.
Its audience is primarily developers of Plone and its projects and add-ons.
```

stevepiercy commented on docs/reference-guide/index.md:
````diff
@@ -0,0 +1,27 @@
+
+This reference guide details functions, modules, and objects included in Plone, describing what they are and what they do. For learning how to use Plone, see the [complete documentation](https://6.docs.plone.org/index.html).
+
+```{note}
+This part of the documentation is under construction.
+You can find documentation for plone in various locations through the [search feature](https://6.docs.plone.org/search.html?q=plone).
+You can help consolidate all of reference material here, even if it is to let us know what is missing, by [creating an issue](https://github.com/plone/documentation/issues/new?assignees=&labels=&projects=&template=new-issue-form.yml).
+```
````
We've tried using "under construction", TODOs, and other things to encourage contributions to documentation, none of which have been effective.

I just created an issue out of disgust and embarrassment https://github.com/plone/documentation/issues/1882.

```suggestion
```

stevepiercy commented on docs/reference-guide/index.md:
````diff
@@ -0,0 +1,27 @@
+You can help consolidate all of reference material here, even if it is to let us know what is missing, by [creating an issue](https://github.com/plone/documentation/issues/new?assignees=&labels=&projects=&template=new-issue-form.yml).
+```
+
+## Plone REST API
+
+[Plone REST API](https://6.docs.plone.org/plone.restapi/docs/source/index.html)
+
+
+## Plone.api
+
+[Plone.api](https://6.docs.plone.org/plone.api/index.html)
````
I think we can put the above two headings and links into a toctree. When we put things into toctrees in different files, it destroys the site navigation, and they already exist elsewhere in the submodules.

However, I've used `sphinx-reredirects` with landing pages to trick it. See for example the source of [Contributing to plone.api](https://6.docs.plone.org/contributing/plone-api.html) and other files in the contributing directory. If you'd like to try hacking that out, please be my guest!

stevepiercy commented on docs/index.md:
````diff
@@ -13,13 +13,65 @@ myst:
+```
+::::
+
+::::{grid-item-card} 
+:link: https://6.docs.plone.org/install/index.html
+{octicon}`zap;1.5em;sd-text-info`
+Get started
+^^^
+This part of the documentation helps you find the best way to get started with Plone, depending on what you want to do.
++++
+```{button-link} https://6.docs.plone.org/install/index.html
+:color: primary
+:expand:
+Go to Get started
+```
+::::
+
````
I think Get Started should be at the start. Let's move this to the first card.

I think we should also add a "User guide" card as the second card. We will need a new page as well, `user-guide/index.md`. It should include a description and two links.

1. https://6.docs.plone.org/volto/user-manual/index.html
2. [Content Editing for Plone](https://training.plone.org/content-editing/index.html).

stevepiercy commented on docs/index.md:
````diff
@@ -13,13 +13,65 @@ myst:
-- Designers who create site themes
-- Deployers who configure servers for site hosting
-- Developers who customize a site’s capabilities, create add-ons, and contribute to Plone itself
+::::{grid-item-card} 
+:link: https://6.docs.plone.org/contributing/index.html
+{octicon}`book;1.5em;sd-text-info`
+Contributing guide
+^^^
+This part of the documentation describes how to contribute to Plone, including all its projects and repositories under the Plone GitHub organization.
++++
+```{button-link} https://6.docs.plone.org/contributing/index.html
+:color: primary
+:expand:
+Go to Contributing guide
+```
+::::
````
Let's move the Contributing guide card to the last position.

stevepiercy commented on docs/index.md:
````diff
@@ -13,13 +13,65 @@ myst:
+```
+::::
+
+::::{grid-item-card}
+:link: https://6.docs.plone.org/developer-guide/index.html
+{octicon}`person;1.5em;sd-text-info`
+Developer guide
+^^^
+This part of the documentation provides information for how to develop in Plone.
++++
+```{button-link} https://6.docs.plone.org/developer-guide/index.html
+:color: primary
+:expand:
+Go to Developer guide
+```
+::::
````
Let's move the Developer guide card to the third position.

Let's add a card for the Reference guide card to the fourth position.

stevepiercy commented on docs/index.md:
````diff
@@ -13,13 +13,65 @@ myst:
+```
+::::
+
+::::{grid-item-card} 
+:link: https://6.docs.plone.org/conceptual-guides/index.html
+{octicon}`light-bulb;1.5em;sd-text-info`
+Conceptual guides 
+^^^
+This part of the documentation provides explanation of concepts to deepen and broaden your understanding of Plone.
++++
+```{button-link} https://6.docs.plone.org/conceptual-guides/index.html
+:color: primary
+:expand:
+Go to Conceptual guides
+```
+::::
+:::::
````
Let's move the Conceptual guides card to the fifth position.

stevepiercy reviewed (changes requested):
@Manas-Kenge thank you so much for taking this on! I'm seeing the future unfold before my eyes, as I envisioned in https://github.com/plone/Products.CMFPlone/issues/4097.

rohnsha0 reviewed (commented):
LGTM

stevepiercy committed:
Merge branch '6.0' into home-page-UI

Manas-Kenge committed:
add additional cards

Manas-Kenge commented:
@stevepiercy I made changes as per your request but user guide's description needs your attention.

davisagli commented on docs/index.md:
```diff
@@ -13,13 +13,94 @@ myst:
+::::
+
+::::{grid-item-card} 
+:link: http://127.0.0.1:8050/user-guide/index.html
```
(This feedback is for all the links on this page.)

We can't use a hardcoded link to localhost or the published docs! They aren't correct when trying to preview this page on https://plone6--1875.org.readthedocs.build/

@stevepiercy Is it possible to use sphinx-style links here?

stevepiercy replied:
All links to a page to internal documentation should use the following pattern.

```
{doc}`path/to/page`
```

The page should not include the file suffix of `.md`. This should render to HTML with correct relative URLs.

stevepiercy replied:
See reference https://6.docs.plone.org/contributing/documentation/myst-reference.html#cross-references

Manas-Kenge replied:
Hey @stevepiercy, I tried using your method for linking, but the problem is that Sphinx Design does not support the {doc} syntax inside it. The link part may work, but the button's styling disappears. Could you try it locally and show me how it works?

stevepiercy replied:
Ah, that's right, there's a special syntax. See https://sphinx-design.readthedocs.io/en/latest/cards.html#clickable-cards.

```
:::{card} Clickable Card (internal)
:link: cards-clickable
:link-type: ref
:link-alt: clickable cards

The entire card can be clicked to navigate to the `cards-clickable` reference target.
:::
```

So for a valid link to a page, you'd do something like this.

```
:link: path/to/page
:link-type: doc
```

davisagli commented on docs/index.md:
```diff
@@ -13,13 +13,94 @@ myst:
+::::{grid-item-card} 
+:link: https://6.docs.plone.org/conceptual-guides/index.html
+{octicon}`light-bulb;1.5em;sd-text-info`
+Conceptual guides 
```
I would put Conceptual guides before Developer guide. It's probably a good idea to understand some concepts before you dive into developing.

stevepiercy replied:
For some conceptual guides, like choosing a user interface or compare buildout and pip, I'd agree. In this case, a how-to guide includes a link to the relevant conceptual guide for deeper understanding if the reader wants to go there. Others require advanced technical knowledge to understand, or describe how sausage is made, and for those I'd disagree.

When I develop, I seek how-tos first, API references second, source code third, and as a last resort conceptual guides. I think that's what most developers do, try stuff out, and when it doesn't work, dig deeper and deeper.

davisagli replied:
@stevepiercy You may be right and I don't feel strongly about this. I'll resolve the comment, and we can always revisit it later.

davisagli commented on docs/index.md:
```diff
@@ -13,13 +13,94 @@ myst:
+::::{grid-item-card} 
+:link: https://6.docs.plone.org/contributing/index.html
+{octicon}`book;1.5em;sd-text-info`
+Contributing guide
```
```suggestion
Contributor guide
```

davisagli commented on docs/index.md:
````diff
@@ -13,13 +13,94 @@ myst:
+```{button-link} https://6.docs.plone.org/contributing/index.html
+:color: primary
+:expand:
+Go to Contributing guide
````
```suggestion
Go to Contributor guide
```

davisagli commented on docs/index.md:
````diff
@@ -13,13 +13,94 @@ myst:
+Go to User guide
+```
+::::
+
````
@stevepiercy Seems to me that the Admin guide should also be mentioned at this same level.

stevepiercy replied:
I regret the distinction of an Admin guide from Developer guide, and I would like to merge them. Admin, Develop, and Deploy have a lot of overlapping requisite knowledge. This also makes it easier to have related content in a single guide and makes it easier to have separation of content along the Diátaxis framework.

davisagli replied:
@stevepiercy I sort of see your point.

Let me express my main concern: The admin guide explains how to run Plone, even if you aren't extending it. "Developer guide" doesn't strike me as an intuitive place to look for that.

Maybe it can work as long as we make it clear that it's for people who are developing a Plone site (not just developing code for a Plone site) and that it includes sysadmin/devops topics as well. The current blurb for the Develop guide card doesn't make this clear. ("This part of the documentation provides information for how to develop in Plone.")

stevepiercy replied:
> The admin guide explains how to run Plone, even if you aren't extending it. 

@davisagli it provides that, and a lot more. 😞 

![Screenshot 2025-03-03 at 9 40 44 PM](https://github.com/user-attachments/assets/31d8cf93-346c-485a-ac61-963d0cb1da4d)

How about this for Develop guide?

"This part of the documentation describes how to install, extend, deploy, maintain, and other technical topics to develop a Plone site."

davisagli replied:
@stevepiercy Well, that certainly covers it, but is a bit verbose. Honestly just "This part of the documentation describes how to develop a Plone site." could work. At first glance that might not look much different from what's already proposed in this PR, but I think "develop a Plone site" expresses a much clearer goal than "develop in Plone"

I don't mean to bikeshed. Let's try something and we'll find out where people still need help with wayfinding. It's already plenty better than the current home page.

Manas-Kenge replied:
@stevepiercy @davisagli What's the final decision? What am I supposed to do?

davisagli replied:
@Manas-Kenge In the end it's a small change: Ignore my suggestion to add the admin guide here. Change the description of the Developer guide to "This part of the documentation describes how to develop a Plone site."

davisagli commented on docs/reference-guide/index.md:
```diff
@@ -0,0 +1,22 @@
+# Reference guide
+
+This reference guide details application programming interfaces (APIs), functions, modules, and objects included in Plone, describing what they are and what they do.
+Its audience is primarily developers of Plone and its projects and add-ons.
```
@stevepiercy I'm not sure this structure is right. Just because a section of docs covers an API doesn't mean that it's all reference. The lists of endpoints / functions are reference but there's also conceptual and how-to material in there.

stevepiercy replied:
There is a way to solve this by ignoring `plone.restapi`'s and `plone.api`'s `index.md` files and using custom ones in `plone/documentation`. It's hard to describe without an example, but it works.

stevepiercy replied:
OK this is now ready for review. Please take a look at the pull request preview.

Manas-Kenge committed:
Merge branch '6.0' into home-page-UI

Manas-Kenge committed:
use internal links for cards

stevepiercy commented:
@Manas-Kenge I move at the speed of unpaid open source software volunteers. Thanks for your work and patience. I will give this a thorough review soon.

stevepiercy commented on docs/index.md:
```diff
@@ -13,13 +13,100 @@ myst:
+{octicon}`person;1.5em;sd-text-info`
+User guide 
+^^^
+This part of the documentation helps you understand how to use Plone effectively, covering essential features, navigation, content management, and common tasks for users and administrators.
```
```suggestion
This part of the documentation helps you understand how to use Plone effectively, covering essential features, navigation, content management, and other common tasks for site editors and administrators.
```

stevepiercy commented on docs/index.md:
```diff
@@ -13,13 +13,100 @@ myst:
+{octicon}`person-add;1.5em;sd-text-info`
+Developer guide
+^^^
+This part of the documentation describes how to develop a Plone site.
```
```suggestion
This part of the documentation describes how to develop a Plone project.
```

stevepiercy commented on docs/index.md:
```diff
@@ -13,13 +13,100 @@ myst:
+{octicon}`bookmark;1.5em;sd-text-info`
+Reference guide 
+^^^
+This part of the documentation provides explanation of APIs, functions, modules, and objects included in Plone, describing what they are and what they do.
```
```suggestion
This part of the documentation describes the APIs, functions, modules, and objects included in Plone.
```

stevepiercy commented on docs/index.md:
```diff
@@ -13,13 +13,100 @@ myst:
+::::{grid-item-card} 
+:link: contributing/index
+:link-type: doc
+{octicon}`book;1.5em;sd-text-info`
```
The book icon represents documentation, whereas the pull request icon relates better to making a contribution.

```suggestion
{octicon}`git-pull-request;1.5em;sd-text-info`
```

stevepiercy commented on docs/reference-guide/index.md:
I'm going to come back to this file to verify my assumptions with a local build of this branch.

stevepiercy commented on docs/user-guide/index.md:
```diff
@@ -0,0 +1,22 @@
+
+# User guide
+
+This user guide details application programming interfaces (APIs), functions, modules, and objects included in Plone, describing what they are and what they do.
+Its audience is primarily developers of Plone and its projects and add-ons.
```
```suggestion
This part of the documentation helps you understand how to use Plone effectively, covering essential features, navigation, content management, and other common tasks for site editors and administrators.
```

stevepiercy commented on docs/user-guide/index.md:
```diff
@@ -0,0 +1,22 @@
+This user guide details application programming interfaces (APIs), functions, modules, and objects included in Plone, describing what they are and what they do.
+Its audience is primarily developers of Plone and its projects and add-ons.
+
+## Plone User Manual
+
+[Plone User Manual](https://6.docs.plone.org/volto/user-manual/index.html)
+
+
+## Content Editing for Plone
+
+[Content Editing for Plone](https://training.plone.org/content-editing/index.html)
```
I'll come back to this file after I test out a few things in a local branch.

stevepiercy commented on submodules/plone.api:
The submodules should not be committed in a PR. I forget how to remove them, and it's always a challenge. Going from a fuzzy memory, I think you have to tell git to point the submodule to a previous commit.

stevepiercy reviewed (changes requested):
This is looking great. I have a few suggestions, and I want to try to build it locally and test out my assumptions about the toctree and `sphinx-reredirects` usage before merging.

jensens commented:
I love it, good work!

Manas-Kenge committed:
modify descriptions

stevepiercy commented on submodules/plone.api:
See this SO post for guidance on how to remove this submodule from the pull request.

https://stackoverflow.com/a/46500544/2214933

stevepiercy replied:
Awesome! That did it!

Manas-Kenge replied:
Thanks for the help! Should I update the branch as well?

stevepiercy replied:
Yes, please, and thank you, as I see you already did so.

stevepiercy reviewed (changes requested):
Everything looks good, except the submodule.

I need to work on the two indices before merging.

Manas-Kenge committed:
restore submodules

Manas-Kenge committed:
Merge branch '6.0' into home-page-UI

jensens commented:
> verything looks good, except the submodule.

Submodules are always a PITA.

stevepiercy commented:
> > verything looks good, except the submodule.
> 
> Submodules are always a PITA.

What would you recommend instead? I'm all ears for simplification.

stevepiercy commented:
Meanwhile, I will add the procedure to a new subsection under https://6.docs.plone.org/contributing/documentation/index.html#edit-in-your-local-editor. "Oops! I committed a submodule!"

stevepiercy committed:
Merge branch '6.0' into home-page-UI

stevepiercy committed:
Merge branch '6.0' into home-page-UI

jensens commented:
> > > verything looks good, except the submodule.
> > 
> > 
> > Submodules are always a PITA.
> 
> What would you recommend instead? I'm all ears for simplification.

Simplification is difficult. Either cloning manually (or have a script to do so) and manual handling of several repos - or submodules. Both is not ideal. First would at least not result in strange commits, where some ghostly submodule changes are involved. 

Or better docs on how to avoid or at least clean up those problem with submodules (do not look a t me, I have no clue - as I always avoided submodules)

stevepiercy committed:
Merge branch '6.0' into home-page-UI

Manas-Kenge committed:
Merge branch '6.0' into home-page-UI

stevepiercy committed:
Merge branch '6.0' into home-page-UI

stevepiercy committed:
Merge branch '6.0' into home-page-UI

stevepiercy committed:
Fix button-ref link syntax and use backtricks instead of colons

stevepiercy committed:
Remove HTML from metadata

stevepiercy committed:
Now that we have cards on the index page, remove redundant sections from `install/index.md`

stevepiercy committed:
Align Get Started card with its content

stevepiercy committed:
Exclude files from plone.restapi to avoid errors

stevepiercy committed:
Shorten text in user guide card

stevepiercy committed:
Fix broken references

stevepiercy committed:
Create bogus editor.md page and add sphinx-reredirect entry to send it to the Volto User Manual

Manas-Kenge committed:
Merge branch '6.0' into home-page-UI

stevepiercy committed:
Fix typo

stevepiercy committed:
- Set up a collection of reference guides as a definition list
- Add a hidden toctree to keep the same list hidden on the page, but appear in the navigation
- Configure the navigation items to redirect to their sources

stevepiercy committed:
Update the name of the navigation item

stevepiercy committed:
Update the user guide index to use a definition list

stevepiercy commented:
This is ready for a final review. I polished up a few minor things. The major changes are with these two sections.

- [User guide](https://plone6--1875.org.readthedocs.build/user-guide/index.html)
- [Reference guide](https://plone6--1875.org.readthedocs.build/reference-guide/index.html)

For both sections:

- Each of the items in the definition list links directly to its authoritative reference documentation.
- Each of these items is in a hidden `toctree` element, so that they don't display in the main content of the page, but they do appear in the navigation tree.
- Each of the items in the navigation tree redirects to the authoritative reference documentation.

stevepiercy committed:
Merge branch '6.0' into home-page-UI

stevepiercy committed:
Merge branch '6.0' into home-page-UI

stevepiercy committed:
Merge branch '6.0' into home-page-UI

stevepiercy committed:
Merge branch '6.0' into home-page-UI

stevepiercy commented:
Last chance to review this PR. I'll merge this weekend, if there are no objections or suggestions for improvement.

stevepiercy committed:
Merge branch '6.0' into home-page-UI
